### PR TITLE
Update formatting configurations and format all files

### PR DESCRIPTION
**Pull request by dengr1065**
- Updated Prettier to handle `import ... with` syntax
- Switched to enforcing LF instead of partially CRLF, partially leaving EOL as-is
- Merged `.gitignore` files into one
- Formatted all code using new configurations
- Excluded potentially large files from formatting by extension (.mp3, .png, etc.)
- Modified `prettier-all` script to format entire workspace according to configurations

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,3 +54,7 @@ config.local.js
 
 build_output
 tmp
+
+src/js/built-temp
+translations/tmp
+gulp/additional_build_files
```
Comment (EmeraldBlock) [inline .gitignore:60]: `tmp` already matches `translations/tmp` (remove the former?); also maybe this stuff should be rearranged to be by the `# Buildfiles` stuff above

**Review by EmeraldBlock — COMMENTED** [review]
skimmed everything and looks fine

File: .gitignore (hunk 1 above)
Comment (dengr1065) [reply .gitignore:60]: As I understand, the root .gitignore was copied from GitHub templates for Node.js project. There are many redundant patterns, and I didn't touch any of these.

Comment (EmeraldBlock) [reply .gitignore:60]: I guess in this aspect this PR is just rearranging things; the temp files' locations need to be revisited in the future probbaly

**Pull request merged**